---
title: "Differential expression analysis"
author: "Alicia Pliego"
date: "3/16/2022"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_collapsed: true
    toc_depth: 3
    number_sections: true
    theme: sandstone
runtime: shiny
---

```{r setup, include=FALSE, echo=FALSE, message=FALSE}
library("DESeq2")
library(tximport)
library(plotly)
library(pheatmap)
library(org.Hs.eg.db)
library(DOSE)
library(pathview)
library(clusterProfiler)
library(AnnotationHub)
library(ensembldb)
library(tidyverse)
  library(EnhancedVolcano)
library(ComplexHeatmap)
library(DT)
```


# Differential expression analysis using DESeq2

```{r,  echo=FALSE, warning=FALSE, message=FALSE}
#Import quantification files generated by DRAGEN from SALMON.
dirs <- list.files("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/data_v2/")
```

```{r,echo=FALSE, warning=FALSE, message=FALSE}
quant_files <- list.files("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/data_v2/",pattern="quant.genes.sf",recursive = TRUE,full.names = TRUE)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# txi import quant files by genes from salmon into DESEQ2
txi <- tximport(quant_files,type="salmon", txOut=TRUE)
```
### Samples to compare in the Differential Analysis
Excluding sample 31 from the analysis, a Differential expression analysis was performed:

```{r,  echo=FALSE, warning=FALSE, message=FALSE,fig.align='center'}
metadata = read.table('/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/BRAF_samples_v2.txt', header = TRUE)
fig <- plot_ly(
  type = 'table',
  header = list(
    values = c("<b>Samples</b>", names(metadata)),
  align = c('left', rep('center', ncol(metadata))),
  line = list(width = 1, color = 'black'),
  fill = list(color = 'rgb(235, 100, 230)'),
  font = list(family = "Arial", size = 14, color = "white")
  ),
  cells = list(
    values = rbind(
      rownames(metadata), 
      t(as.matrix(unname(metadata)))
    ),
    align = c('left', rep('center', ncol(metadata))),
    line = list(color = "black", width = 1),
    fill = list(color = c('rgb(235, 193, 238)', 'rgba(228, 222, 249, 0.65)')),
    font = list(family = "Arial", size = 12, color = c("black"))
  ))

fig
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeqDataSetFromTximport(txi, 
                                colData = metadata,
                                design <- ~Treatment)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds) >= 10
keep <- rowSums(assay(dds) >= 10) >= 2
dds <- dds[keep,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeq(dds)
res <- results(dds, contrast = c("Treatment", "Treated", "Untreated"))
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
resLFC <- lfcShrink(dds, coef= "Treatment_Untreated_vs_Treated", type="apeglm")
```
## Comparison treated vs untreated groups 
Summary of differentially expressed genes:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
resOrdered <- res[order(res$pvalue),]
summary(res)
```

Summary of differentially expressed genes.
Adjusting cut-off for and FDR of 0.05:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05 <- results(dds, alpha=0.05, contrast = c("Treatment", "Treated", "Untreated"))
summary(res05)
```

Cutoff values:
  pvalue  = 10e-16
  FCcutoff = 1.5

### Volcano plot 
Genes upregulated in the treated sample are in the right of the volcano plot, whilst downregulated genes are in the left hand side.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
EnhancedVolcano(res05,
    lab = rownames(res05),
    title = 'Treated vs Untreated',
    pCutoff = 10e-16,
    FCcutoff = 1.5,
    x = 'log2FoldChange',
    y = 'pvalue')

```

``````{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}

  EnhancedVolcano(res05,
    lab = rownames(res05),
    x = 'log2FoldChange',
    y = 'pvalue',
    selectLab = c('DLL1','CCND2','NOTCH1',
      'TP53','LEF1','SKP2','MYC','CUL1', 'HDAC2', 'PYGO1','AMER1', 'TCF7'),
    xlab = bquote(~Log[2]~ 'fold change'),
    pCutoff = 10e-14,
    FCcutoff = 2.0,
    pointSize = 4.0,
    labSize = 6.0,
    labCol = 'black',
    labFace = 'bold',
    boxedLabels = TRUE,
    colAlpha = 4/5,
    legendPosition = 'right',
    legendLabSize = 14,
    legendIconSize = 4.0,
    drawConnectors = TRUE,
    widthConnectors = 1.0,
    colConnectors = 'black')
```


### Interactive volcano plot
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05_df <- as.data.frame(res05)
res05_df$log10P <- -log10(res05_df$pvalue) 
```



```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
library(plotly)

vline <- function(x = 0, color = "red") {
  list(
    type = "line", 
    y0 = 0, 
    y1 = 1, 
    yref = "paper",
    x0 = x, 
    x1 = x, 
    line = list(color = color)
  )
}
hline <- function(y = 0, color = "blue") {
  list(
    type = "line", 
    x0 = 0, 
    x1 = 1, 
    xref = "paper",
    y0 = y, 
    y1 = y, 
    line = list(color = color)
  )
}


fig <- plot_ly(data = res05_df, x = ~res05_df$log2FoldChange, y = ~res05_df$log10P,
               text= rownames(res05_df), color = ~res05_df$log2FoldChange) %>%
  layout(xaxis = list(title = 'log2FoldChange'), font=t, plot_bgcolor = "#e5ecf6",
         yaxis = list(title = '-log10(Pval)'), legend = list(title=list(text='Legend Title')), shapes = list(vline(2.5), hline(20)))

fig 
```

### Counts of most differentially expressed gene

Expression of MYC and GPRC5B is significantly higher in the Treated group.

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
d <- plotCounts(dds, gene="GATA6", intgroup="Treatment", returnData=TRUE)

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = Treatment, y = count, color = Treatment)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("GATA6") +
  theme(plot.title = element_text(hjust = 0.5))
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
par(mfrow=c(2,3))

d <- plotCounts(dds, gene="GPRC5B", intgroup="Treatment", returnData=TRUE)

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = Treatment, y = count, color = Treatment)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("GPRC5B") +
  theme(plot.title = element_text(hjust = 0.5))


```

### Counts of other Differenatially expressed genes
Other genes of interest such as  B-Catenin (CTNNB1) abd TP53 are also upregulated in the treated group.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(2,3))

plotCounts(dds, gene="CTNNB1", intgroup="Treatment")
plotCounts(dds, gene="MYC", intgroup="Treatment")
plotCounts(dds, gene="TP53", intgroup="Treatment")
plotCounts(dds, gene="ALDOB", intgroup="Treatment")
plotCounts(dds, gene="CDH12", intgroup="Treatment")
plotCounts(dds, gene="DLL1", intgroup="Treatment")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Further filter DE genes based on Padj = 0.01,  and FC>1.5(L2FC = 0.58)
res_filtered_pval <- subset(res05, padj <0.01)
res_filtered_all_up <- subset(res_filtered_pval, log2FoldChange >0.58)
res_filtered_all_up_df <- as.data.frame(res_filtered_all_up)

res_filtered_all_down <- subset(res_filtered_pval, log2FoldChange < -0.58)
res_filtered_all_down_df <- as.data.frame(res_filtered_all_down)
```

### Table of downregulated genes in Treated samples

```{r, echo=FALSE}
library(shiny)
library(DT)
```

```{r, echo=FALSE}
DT::datatable(res_filtered_all_down_df)
```

### Table of upregulated genes in Treated samples
```{r, echo=FALSE}
DT::datatable(res_filtered_all_up_df)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Rank genes and export for GSEA analysis
x <- res_filtered_pval
x$fcsign <- sign(x$log2FoldChange )
x$logP=-log10(x$pvalue)
x$metric= x$logP/x$fcsign
x$Gene <- rownames(x)
y<-x[,c("Gene", "metric")]
filtered <- na.omit(y)
write.table(filtered ,file="DE_genes.rnk",quote=F,sep="\t",row.names=F)
```

## Heatmap of top 50 DE genes

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Select TOP 20 DE genes by pvalue
res_df<- as.data.frame(res_filtered_pval)
resOrdered   <- res_df[order(res_df$pvalue),]
resOrdered50 <- resOrdered[1:50,]
top50genes   <- rownames(resOrdered50)
top50genes   <- as.data.frame(top50genes)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Normalize results
dds <- estimateSizeFactors(dds)
normalized_counts <- counts(dds, normalized=T) %>% 
                     data.frame() %>%
                     rownames_to_column(var="gene") 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

## This will bring in a column of gene symbols
top50_counts<- merge(normalized_counts, top50genes, by.x="gene", by.y="top50genes")
names(top50_counts)[2] <- "LR2022_25"
names(top50_counts)[3] <- "LR2022_26"
names(top50_counts)[4] <- "LR2022_27"
names(top50_counts)[5] <- "LR2022_28"
names(top50_counts)[6] <- "LR2022_29"
names(top50_counts)[7] <- "LR2022_30"
names(top50_counts)[8] <- "LR2022_32"
names(top50_counts)[9] <- "LR2022_33"
names(top50_counts)[10] <- "LR2022_34"
names(top50_counts)[11] <- "LR2022_35"
names(top50_counts)[12] <- "LR2022_36"
names(top50_counts)[13] <- "LR2022_37"
top50_counts2 <- top50_counts[,-1]
rownames(top50_counts2) <- top50_counts[,1]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

vst <- vst(dds, blind=FALSE)
vst <- assay(vst)
vst <- as.data.frame(vst)
vst_sig <- vst[rownames(vst) %in% rownames(top50_counts2),]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
names(vst_sig)[1] <- "LR2022_25"
names(vst_sig)[2] <- "LR2022_26"
names(vst_sig)[3] <- "LR2022_27"
names(vst_sig)[4] <- "LR2022_28"
names(vst_sig)[5] <- "LR2022_29"
names(vst_sig)[6] <- "LR2022_30"
names(vst_sig)[7] <- "LR2022_32"
names(vst_sig)[8] <- "LR2022_33"
names(vst_sig)[9] <- "LR2022_34"
names(vst_sig)[10] <- "LR2022_35"
names(vst_sig)[11] <- "LR2022_36"
names(vst_sig)[12] <- "LR2022_37"
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- t(scale(t(vst_sig)))
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 14, fig.align='center'}
Heatmap(heat, 
      row_names_gp = gpar(fontsize = 8),
    top_annotation = HeatmapAnnotation(foo = anno_block(gp = gpar(fill = 2:4),
        labels = c("Treated", "Untreated"), 
        labels_gp = gpar(col = "white", fontsize = 10))),
    column_km = 2)
```


## GSEA results
Gene Set Enrichment analysis of Differential expressed genes filtered by pval <0.01
Enriched pathways with a pval <0.05:
```{r pressure, echo=FALSE, out.width = '100%'}
knitr::include_graphics("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/report/RNAseq-BRAF/images/GSEA_plot.png")
```

Link to enrichment results:

Read more [here](GSEA_180322/index.html) 


# Cluster Profiles

## GO Enrichment

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# we want the log2 fold change 
original_gene_list = res2$log2FoldChange
#original_gene_list<- as.data.frame(original_gene_list)
# name the vector
names(original_gene_list) <- rownames(res2)

# omit any NA values 
gene_list<-na.omit(original_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)

gene <- names(gene_list)[abs(gene_list) > 2]


gse <- gseGO(geneList=gene_list, 
             ont ="BP", 
             keyType = "SYMBOL",
             minGSSize = 10, 
             maxGSSize = 800, 
             pvalueCutoff = 0.01, 
             verbose = TRUE, 
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "BH")
```


```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 10, fig.align='center'}

dotplot(gse, showCategory = 10, font.size=14, split=".sign") + facet_grid(.~.sign)
```

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 10, fig.align='center'}

#require(DOSE)
#dotplot(gse, showCategory=3, split=".sign") + facet_grid(.~.sign)

```

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}

# Use the `Gene Set` param for the index in the title, and as the value for geneSetId
#gseaplot(gse, by = "all", title = gse$Description[6], geneSetID = 1)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

# Convert gene IDs for gseKEGG function
# We will lose some genes here because not all IDs will be converted
ids = bitr(names(original_gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb = org.Hs.eg.db)
```

#### GSEA KEGG pathways with Cluster Profiler

The normalized enrichment score (NES) is the primary statistic for examining gene set enrichment results. By normalizing the enrichment score, GSEA accounts for differences in gene set size and in correlations between gene sets and the expression dataset; therefore, the normalized enrichment scores (NES) can be used to compare analysis results across gene sets.

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# remove duplicate IDS (here we use "SYMBOL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

# Create a new dataframe df2 which has the respective entrez IDs for the gene symbols.
colnames(dedup_ids) = c("GeneSymbol", "EntrezID")
res$GeneSymbol <- rownames(res)
res2 <- as.data.frame(res)
df2 = merge(res2, dedup_ids, by = "GeneSymbol")

# Create a vector of the gene universe
kegg_gene_list = df2$log2FoldChange

# Name vector with ENTREZ ids
names(kegg_gene_list) = df2$EntrezID

# omit any NA values 
kegg_gene_list = na.omit(kegg_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "hsa"
kk2 = gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

kk3 = gseMKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05)
```

Plot results from KEGG pathways 
```{r, echo=FALSE}
out_kk3 = as.data.frame(kk2@result)
DT::datatable(out_kk3)
```

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 10, fig.align='center'}

dotplot(kk2, showCategory = 15, title = "Enriched Pathways",font.size=14, split=".sign") + facet_grid(.~.sign)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
out = as.data.frame(kk2@result)
#out = out[, 1:30]
#head(out)
```

Plot the results from the enrichment as a table
```{r, echo=FALSE}
DT::datatable(out)
```
Plot the ENTREZ ID to Genesymbol table
```{r, echo=FALSE}
DT::datatable(df2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
library(msigdbr)
m_df <- msigdbr(species = "Homo sapiens")

m_t2g <- msigdbr(species = "Homo sapiens", category = "C6") %>% 
  dplyr::select(gs_name, entrez_gene)


```

### Oncogenic pathways
```{r, echo=FALSE, warning=FALSE, message=FALSE}
C6_t2g <- msigdbr(species = "Homo sapiens", category = "C6") %>% 
  dplyr::select(gs_name, entrez_gene)
#head(C6_t2g)
em2 <- GSEA(kegg_gene_list, TERM2GENE = C6_t2g)
out_em2 = as.data.frame(em2@result)
#head(em2)
```

```{r, echo=FALSE}
DT::datatable(out_em2)
```

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 10, fig.align='center'}
dotplot(em2, showCategory = 15,title = "Enriched Pathways" ,font.size=14, split=".sign") + facet_grid(.~.sign)
```

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}

gseaplot(em2, by = "all", title = em2$Description[1], geneSetID = 1)

```

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}

#gseaplot(kk2, by = "all", title = kk2$Description[13], geneSetID = 1)

```

```{r enrichWP, echo=FALSE, warning=FALSE, message=FALSE}

gene <- names(kegg_gene_list)[abs(kegg_gene_list) > 2]

enrichWP(gene, organism = "Homo sapiens") 
```

```{r em3, echo=FALSE, warning=FALSE, message=FALSE}
em3 <- gseWP(kegg_gene_list, organism = "Homo sapiens")
```

```{r dotplot em3 , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 10, fig.align='center'}
dotplot(em3, title = "Enriched Pathways",font.size=14, split=".sign") + facet_grid(.~.sign)
```

```{r reactomePA, echo=FALSE, warning=FALSE, message=FALSE}
#library(ReactomePA)
#x <- enrichPathway(gene=de, pvalueCutoff = 0.05, readable=TRUE)

#y <- gsePathway(geneList, 
#                pvalueCutoff = 0.2,
#                pAdjustMethod = "BH", 
#                verbose = FALSE)
#head(y)
```

```{r viepathway, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 10, fig.align='center'}

#viewPathway("E2F mediated regulation of DNA replication", 
#            readable = TRUE, 
#            foldChange = geneList)
```




