---
title: "Differential expression analysis"
author: "Alicia Pliego"
date: "3/16/2022"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_collapsed: true
    toc_depth: 3
    number_sections: true
    theme: sandstone

---

```{r setup, include=FALSE, echo=FALSE, message=FALSE}
library("DESeq2")
library(tximport)
library(plotly)
library(pheatmap)
library(org.Hs.eg.db)
library(DOSE)
library(pathview)
library(clusterProfiler)
library(AnnotationHub)
library(ensembldb)
library(tidyverse)
  library(EnhancedVolcano)
library(ComplexHeatmap)
library(DT)
```

# Differential expression analysis using DESeq2

```{r,  echo=FALSE, warning=FALSE, message=FALSE}
#Import quantification files generated by DRAGEN from SALMON.
#dirs <- list.files("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/data_v2/")
```

```{r,echo=FALSE, warning=FALSE, message=FALSE}
quant_files2 <- list.files("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/data_v4/",pattern="quant.genes.sf",recursive = TRUE,full.names = TRUE)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# txi import quant files by genes from salmon into DESEQ2
txi2 <- tximport(quant_files2,type="salmon", txOut=TRUE)
```
### Samples to compare in the Differential Analysis
Excluding sample 31, 35 and 37 from the analysis, a Differential expression analysis was performed:

```{r,  echo=FALSE, warning=FALSE, message=FALSE,fig.align='center'}
metadata2 = read.table('/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/BRAF_samples_v4.txt', header = TRUE)
fig <- plot_ly(
  type = 'table',
  header = list(
    values = c("<b>Samples</b>", names(metadata2)),
  align = c('left', rep('center', ncol(metadata2))),
  line = list(width = 1, color = 'black'),
  fill = list(color = 'rgb(235, 100, 230)'),
  font = list(family = "Arial", size = 14, color = "white")
  ),
  cells = list(
    values = rbind(
      rownames(metadata2), 
      t(as.matrix(unname(metadata2)))
    ),
    align = c('left', rep('center', ncol(metadata2))),
    line = list(color = "black", width = 1),
    fill = list(color = c('rgb(235, 193, 238)', 'rgba(228, 222, 249, 0.65)')),
    font = list(family = "Arial", size = 12, color = c("black"))
  ))

fig
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds_2 <- DESeqDataSetFromTximport(txi2, 
                                colData = metadata2,
                                design <- ~Treatment)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds_2) >= 10
keep <- rowSums(assay(dds_2) >= 10) >= 2
dds_2 <- dds_2[keep,]
```

```{r results , echo=FALSE, warning=FALSE, message=FALSE}
dds_2 <- DESeq(dds_2)
res <- results(dds_2, contrast = c("Treatment", "Treated", "Untreated"))
```


## Comparison treated vs untreated groups 
Summary of differentially expressed genes:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
resOrdered <- res[order(res$pvalue),]
summary(res)
```

Summary of differentially expressed genes.
Adjusting cut-off for and FDR of 0.01:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05 <- results(dds_2, alpha=0.05, contrast = c("Treatment", "Treated", "Untreated"))
summary(res05)
```

Cutoff values:
  pvalue  = 10e-16
  FCcutoff = 1.5

### Volcano plot 
Genes upregulated in the treated sample are in the right of the volcano plot, whilst downregulated genes are in the left hand side.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
EnhancedVolcano(res05,
    lab = rownames(res05),
    title = 'Treated vs Untreated',
    pCutoff = 10e-22,
    FCcutoff = 2,
    x = 'log2FoldChange',
    y = 'pvalue')

```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}

  EnhancedVolcano(res05,
    lab = rownames(res05),
    x = 'log2FoldChange',
    y = 'pvalue',
    selectLab = c('DLL1','CCND2','NOTCH1',
      'TP53','LEF1','SKP2','MYC','CUL1', 'HDAC2', 'PYGO1','AMER1', 'TCF7', 'DKK3', 'GSK3B',
      drawConnectors = TRUE,
    widthConnectors = 0.75,
    xlab = bquote(~Log[2]~ 'fold change'),
    pCutoff = 10e-22,
    FCcutoff = 2.0,
    pointSize = 4.0,
    labSize = 6.0,
    labCol = 'black',
    labFace = 'bold',
    boxedLabels = TRUE,
    colAlpha = 4/5,
    legendPosition = 'right',
    legendLabSize = 14,
    legendIconSize = 4.0,
    drawConnectors = TRUE,
    widthConnectors = 1.0,
    colConnectors = 'black'))
```


### Interactive volcano plot
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05_df <- as.data.frame(res05)
res05_df$log10P <- -log10(res05_df$pvalue) 
```



```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
library(plotly)

vline <- function(x = 0, color = "red") {
  list(
    type = "line", 
    y0 = 0, 
    y1 = 1, 
    yref = "paper",
    x0 = x, 
    x1 = x, 
    line = list(color = color)
  )
}
hline <- function(y = 0, color = "blue") {
  list(
    type = "line", 
    x0 = 0, 
    x1 = 1, 
    xref = "paper",
    y0 = y, 
    y1 = y, 
    line = list(color = color)
  )
}


fig <- plot_ly(data = res05_df, x = ~res05_df$log2FoldChange, y = ~res05_df$log10P,
               text= rownames(res05_df), color = ~res05_df$log2FoldChange) %>%
  layout(xaxis = list(title = 'log2FoldChange'), font=t, plot_bgcolor = "#e5ecf6",
         yaxis = list(title = '-log10(Pval)'), legend = list(title=list(text='Legend Title')), shapes = list(vline(2.5), hline(20)))

fig 
```

### Counts of most differentially expressed gene


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
d <- plotCounts(dds_2, gene="PTEN", intgroup="Treatment", returnData=TRUE)

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = Treatment, y = count, color = Treatment)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("PTEN") +
  theme(plot.title = element_text(hjust = 0.5))
```

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
d <- plotCounts(dds_2, gene="WNT5A", intgroup="Treatment", returnData=TRUE)

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = Treatment, y = count, color = Treatment)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("WNT5A") +
  theme(plot.title = element_text(hjust = 0.5))
```


```{r plotcounts ctnnb1, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
par(mfrow=c(2,3))

d <- plotCounts(dds_2, gene="CTNNB1", intgroup="Treatment", returnData=TRUE)

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = Treatment, y = count, color = Treatment)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("CTNNB1") +
  theme(plot.title = element_text(hjust = 0.5))


```

### Counts of other Differenatially expressed genes
There are genes related to the NOTCH signaling pathway through the DLL1 activation upregulated in the treated group, whilst the other routes of NOTCH throug JAG1 JAG2 are upregulated in the non treated samples.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(2,3))

plotCounts(dds_2, gene="NOTCH1", intgroup="Treatment")
plotCounts(dds_2, gene="RHOA", intgroup="Treatment")
plotCounts(dds_2, gene="WNT3", intgroup="Treatment")
plotCounts(dds_2, gene="TP53", intgroup="Treatment")
plotCounts(dds_2, gene="NOTCH2", intgroup="Treatment")
plotCounts(dds_2, gene="SMAD4", intgroup="Treatment")
```

- Posible compensatory mechanism:

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6387042/

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(1,3))

plotCounts(dds_2, gene="FZD4", intgroup="Treatment")
plotCounts(dds_2, gene="GATA6", intgroup="Treatment")
plotCounts(dds_2, gene="PORCN", intgroup="Treatment")
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(2,3))

plotCounts(dds_2, gene="NOTCH1", intgroup="Treatment")
plotCounts(dds_2, gene="NOTCH2", intgroup="Treatment")
plotCounts(dds_2, gene="NBR2", intgroup="Treatment")
plotCounts(dds_2, gene="DLL1", intgroup="Treatment")
plotCounts(dds_2, gene="DLL4", intgroup="Treatment")
plotCounts(dds_2, gene="NOTCH3", intgroup="Treatment")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(2,3))

plotCounts(dds_2, gene="NOTCH4", intgroup="Treatment")
plotCounts(dds_2, gene="JAG1", intgroup="Treatment")
plotCounts(dds_2, gene="JAG2", intgroup="Treatment")
plotCounts(dds_2, gene="DLL1", intgroup="Treatment")
plotCounts(dds_2, gene="DLL4", intgroup="Treatment")
plotCounts(dds_2, gene="DKK1", intgroup="Treatment")
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(2,3))

plotCounts(dds_2, gene="MALAT1", intgroup="Treatment")
plotCounts(dds_2, gene="APC", intgroup="Treatment")
plotCounts(dds_2, gene="SMAD3", intgroup="Treatment")
plotCounts(dds_2, gene="SMAD4", intgroup="Treatment")
plotCounts(dds_2, gene="SMAD2", intgroup="Treatment")
plotCounts(dds_2, gene="PTEN", intgroup="Treatment")
```

-Alternative compensatory mechanism:
  Down regulation of PTEN which is an inhibitor of the mTOR pathway may lead to an upregulation of MTOR and cell proliferation.

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(2,3))

plotCounts(dds_2, gene="GSK3B", intgroup="Treatment")
plotCounts(dds_2, gene="PIK3CA", intgroup="Treatment")
plotCounts(dds_2, gene="PTEN", intgroup="Treatment")
plotCounts(dds_2, gene="RPTOR", intgroup="Treatment")
plotCounts(dds_2, gene="IKBKB", intgroup="Treatment")

```

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2418585/

It has been shown in prostate cancer that downregulation of PTEN might lead to mTOR pathway downstream from Akt controls NF-kB. In the clones, we can see duplications of NFKB
, and enrichment in the NF-KB pathway.

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(1,3))

plotCounts(dds_2, gene="AKT1", intgroup="Treatment")
plotCounts(dds_2, gene="MTOR", intgroup="Treatment")
plotCounts(dds_2, gene="NFATC1", intgroup="Treatment")
```

- Additionally, WNT pathway we can see an upregulation of Frizzle genes and WNT5A and a downregulation of NFAT gene.
```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Further filter DE genes based on Padj = 0.01,  and FC>1.5(L2FC = 0.58)
res_filtered_pval <- subset(res05, padj <0.01)
res_filtered_all_up <- subset(res_filtered_pval, log2FoldChange >0.58)
res_filtered_all_up_df <- as.data.frame(res_filtered_all_up)

res_filtered_all_down <- subset(res_filtered_pval, log2FoldChange < -0.58)
res_filtered_all_down_df <- as.data.frame(res_filtered_all_down)
```

### Table of downregulated genes in Treated samples

```{r, echo=FALSE}
library(shiny)
library(DT)
```

```{r, echo=FALSE}
DT::datatable(res_filtered_all_down_df)
```

### Table of upregulated genes in Treated samples
```{r, echo=FALSE}
DT::datatable(res_filtered_all_up_df)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Rank genes and export for GSEA analysis
x <- res
x$fcsign <- sign(x$log2FoldChange )
x$logP=-log10(x$pvalue)
x$metric= x$logP/x$fcsign
x$Gene <- rownames(x)
y<-x[,c("Gene", "metric")]
filtered <- na.omit(y)
filtered <- filtered[order((filtered$metric), decreasing = TRUE), ]
#write.table(filtered ,file="/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/gsea/DE_genes_190522_AP_v2.rnk",quote=F,sep="\t",row.names=F)
```

## Heatmap of top 50 DE genes

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Select TOP 20 DE genes by pvalue
res_df<- as.data.frame(res_filtered_pval)
resOrdered   <- res_df[order(res_df$pvalue),]
resOrdered50 <- resOrdered[1:50,]
top50genes   <- rownames(resOrdered50)
top50genes   <- as.data.frame(top50genes)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Normalize results
dds <- estimateSizeFactors(dds_2)
normalized_counts <- counts(dds, normalized=T) %>% 
                     data.frame() %>%
                     rownames_to_column(var="gene") 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

## This will bring in a column of gene symbols
top50_counts<- merge(normalized_counts, top50genes, by.x="gene", by.y="top50genes")

names(top50_counts)[1] <- "LR2022_26"
names(top50_counts)[2] <- "LR2022_27"
names(top50_counts)[3] <- "LR2022_28"
names(top50_counts)[4] <- "LR2022_29"
names(top50_counts)[5] <- "LR2022_30"
names(top50_counts)[6] <- "LR2022_32"
names(top50_counts)[7] <- "LR2022_33"
names(top50_counts)[8] <- "LR2022_34"
names(top50_counts)[9] <- "LR2022_35"
names(top50_counts)[10] <- "LR2022_36"

top50_counts2 <- top50_counts[,-1]
rownames(top50_counts2) <- top50_counts[,1]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

vst <- vst(dds, blind=FALSE)
vst <- assay(vst)
vst <- as.data.frame(vst)
vst_sig <- vst[rownames(vst) %in% rownames(top50_counts2),]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

names(vst_sig)[1] <- "LR2022_26"
names(vst_sig)[2] <- "LR2022_27"
names(vst_sig)[3] <- "LR2022_28"
names(vst_sig)[4] <- "LR2022_29"
names(vst_sig)[5] <- "LR2022_30"
names(vst_sig)[6] <- "LR2022_32"
names(vst_sig)[7] <- "LR2022_33"
names(vst_sig)[8] <- "LR2022_34"
names(vst_sig)[9] <- "LR2022_35"
names(vst_sig)[10] <- "LR2022_36"

```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- t(scale(t(vst_sig)))
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 14, fig.align='center'}
Heatmap(heat, 
      row_names_gp = gpar(fontsize = 8),
    top_annotation = HeatmapAnnotation(foo = anno_block(gp = gpar(fill = 2:4),
        labels = c("Treated", "Untreated"), 
        labels_gp = gpar(col = "white", fontsize = 10))),
    column_km = 2)
```


## GSEA results 5 vs 5 samples 

Gene Set Enrichment analysis of Differential expressed genes filtered by pval <0.01
Enriched pathways with a pval <0.05:
### Hallmarks
```{r , echo=FALSE, out.width = '100%'}
knitr::include_graphics("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/report/RNAseq-BRAF/images/GSEA_plot_240522.png")
```

Link to enrichment results:

Read more [here](new_GSEA_hallmarks.GseaPreranked.1652823888329/index.html) 

### KEGG
```{r , echo=FALSE, out.width = '100%'}
knitr::include_graphics("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/report/RNAseq-BRAF/images/GSEA_plot_KEGG_240522.png")
```

Link to enrichment results:

Read more [here](new_GSEA_kegg.GseaPreranked.1652823697095/index.html) 

