<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Alicia Pliego" />


<title>Enrichment analysis</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.23/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RNAseq-BRAF</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="oncoprint.html">Oncoprint</a>
</li>
<li>
  <a href="de_analysis.html">Clustering analysis</a>
</li>
<li>
  <a href="differential_expression_BRAF.html">Differential expression</a>
</li>
<li>
  <a href="gsea.html">Gene Set enrichment</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Enrichment analysis</h1>
<h4 class="author">Alicia Pliego</h4>

</div>


<div id="gene-enrichment-analysis" class="section level1" number="1">
<h1 number="1"><span class="header-section-number">1</span> Gene Enrichment analysis</h1>
<p>Samples to compare in the Differential Analysis: Excluding sample 31, 35 and 37 from the analysis, a Differential expression analysis was performed:</p>
<div id="wnt-signaling-pathway-enrichment" class="section level2" number="1.1">
<h2 number="1.1"><span class="header-section-number">1.1</span> WNT Signaling pathway enrichment</h2>
<div id="volcano-plot-of-the-de-genes-belonging-to-the-wnt-pathway" class="section level3" number="1.1.1">
<h3 number="1.1.1"><span class="header-section-number">1.1.1</span> Volcano plot of the DE genes belonging to the WNT pathway</h3>
<p><img src="gsea_files/figure-html/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>We can see an enrichment in the treated BRAF cases (right) for positive regulators of the WNT signaling cascade, whilst on the left (downregulated for the treated samples) we can see an enrichment for the suppressor genes. From these results we can say it seems there is an upregulation of activator genes in the treated samples and downregulation of suppressor genes of the WNT signaling.</p>
</div>
<div id="heatmap-of-the-genes-belonging-to-the-wnt-pathway" class="section level3" number="1.1.2">
<h3 number="1.1.2"><span class="header-section-number">1.1.2</span> Heatmap of the genes belonging to the WNT pathway</h3>
<p><img src="gsea_files/figure-html/unnamed-chunk-12-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>Create the named and sorted vector of t-statistics (i.e. for the geneList argument of the gseGO() function).</p>
<p>List of GO terms differentially upregulated:</p>
<p><img src="gsea_files/figure-html/unnamed-chunk-17-1.png" width="768" style="display: block; margin: auto;" /></p>
<p><img src="gsea_files/figure-html/unnamed-chunk-18-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gsea" class="section level2" number="1.2">
<h2 number="1.2"><span class="header-section-number">1.2</span> GSEA</h2>
<div id="pathways" class="section level3" number="1.2.1">
<h3 number="1.2.1"><span class="header-section-number">1.2.1</span> Pathways</h3>
<p><img src="images/hsa04310.pathview.png" width="100%" /> <img src="images/hsa04110.pathview.png" width="100%" /> <img src="images/hsa03008.pathview.png" width="100%" /> <img src="images/hsa04115.pathview.png" width="100%" /> <img src="images/hsa04330.pathview.png" width="100%" /> <img src="images/hsa04010.pathview.png" width="100%" /></p>
<p><img src="images/hsa04310.pathview.png" width="100%" /></p>
</div>
</div>
<div id="hallmarks-msigdb-enriched-pathways" class="section level2" number="1.3">
<h2 number="1.3"><span class="header-section-number">1.3</span> Hallmarks MSigDB enriched pathways</h2>
<div id="htmlwidget-de61d9bae478163ca046" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-de61d9bae478163ca046">{"x":{"filter":"none","vertical":false,"data":[["HALLMARK_E2F_TARGETS","HALLMARK_MYC_TARGETS_V1","HALLMARK_G2M_CHECKPOINT","HALLMARK_OXIDATIVE_PHOSPHORYLATION","HALLMARK_MYC_TARGETS_V2","HALLMARK_DNA_REPAIR","HALLMARK_FATTY_ACID_METABOLISM","HALLMARK_UV_RESPONSE_UP","HALLMARK_MTORC1_SIGNALING","HALLMARK_HYPOXIA","HALLMARK_INTERFERON_GAMMA_RESPONSE","HALLMARK_INTERFERON_ALPHA_RESPONSE","HALLMARK_MITOTIC_SPINDLE","HALLMARK_MYOGENESIS","HALLMARK_COAGULATION","HALLMARK_TGF_BETA_SIGNALING","HALLMARK_CHOLESTEROL_HOMEOSTASIS","HALLMARK_UV_RESPONSE_DN","HALLMARK_APICAL_JUNCTION","HALLMARK_GLYCOLYSIS","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_SPERMATOGENESIS","HALLMARK_ADIPOGENESIS","HALLMARK_INFLAMMATORY_RESPONSE","HALLMARK_WNT_BETA_CATENIN_SIGNALING","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION","HALLMARK_COMPLEMENT","HALLMARK_HEME_METABOLISM","HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY","HALLMARK_PI3K_AKT_MTOR_SIGNALING","HALLMARK_ANDROGEN_RESPONSE"],["HALLMARK_E2F_TARGETS","HALLMARK_MYC_TARGETS_V1","HALLMARK_G2M_CHECKPOINT","HALLMARK_OXIDATIVE_PHOSPHORYLATION","HALLMARK_MYC_TARGETS_V2","HALLMARK_DNA_REPAIR","HALLMARK_FATTY_ACID_METABOLISM","HALLMARK_UV_RESPONSE_UP","HALLMARK_MTORC1_SIGNALING","HALLMARK_HYPOXIA","HALLMARK_INTERFERON_GAMMA_RESPONSE","HALLMARK_INTERFERON_ALPHA_RESPONSE","HALLMARK_MITOTIC_SPINDLE","HALLMARK_MYOGENESIS","HALLMARK_COAGULATION","HALLMARK_TGF_BETA_SIGNALING","HALLMARK_CHOLESTEROL_HOMEOSTASIS","HALLMARK_UV_RESPONSE_DN","HALLMARK_APICAL_JUNCTION","HALLMARK_GLYCOLYSIS","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_SPERMATOGENESIS","HALLMARK_ADIPOGENESIS","HALLMARK_INFLAMMATORY_RESPONSE","HALLMARK_WNT_BETA_CATENIN_SIGNALING","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION","HALLMARK_COMPLEMENT","HALLMARK_HEME_METABOLISM","HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY","HALLMARK_PI3K_AKT_MTOR_SIGNALING","HALLMARK_ANDROGEN_RESPONSE"],["HALLMARK_E2F_TARGETS","HALLMARK_MYC_TARGETS_V1","HALLMARK_G2M_CHECKPOINT","HALLMARK_OXIDATIVE_PHOSPHORYLATION","HALLMARK_MYC_TARGETS_V2","HALLMARK_DNA_REPAIR","HALLMARK_FATTY_ACID_METABOLISM","HALLMARK_UV_RESPONSE_UP","HALLMARK_MTORC1_SIGNALING","HALLMARK_HYPOXIA","HALLMARK_INTERFERON_GAMMA_RESPONSE","HALLMARK_INTERFERON_ALPHA_RESPONSE","HALLMARK_MITOTIC_SPINDLE","HALLMARK_MYOGENESIS","HALLMARK_COAGULATION","HALLMARK_TGF_BETA_SIGNALING","HALLMARK_CHOLESTEROL_HOMEOSTASIS","HALLMARK_UV_RESPONSE_DN","HALLMARK_APICAL_JUNCTION","HALLMARK_GLYCOLYSIS","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_SPERMATOGENESIS","HALLMARK_ADIPOGENESIS","HALLMARK_INFLAMMATORY_RESPONSE","HALLMARK_WNT_BETA_CATENIN_SIGNALING","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION","HALLMARK_COMPLEMENT","HALLMARK_HEME_METABOLISM","HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY","HALLMARK_PI3K_AKT_MTOR_SIGNALING","HALLMARK_ANDROGEN_RESPONSE"],[200,200,198,200,58,149,145,143,198,186,180,94,199,145,108,54,71,134,168,188,190,83,182,165,39,156,177,173,48,98,95],[0.741627358994214,0.70064619437171,0.688427589403065,0.618625040474659,0.772154106865466,0.458123552246206,0.459820155548379,0.454211099417005,0.406188954175867,-0.38693502370746,-0.373945387875108,-0.436232475760224,0.387269918045906,-0.376634062430774,-0.38555317747053,-0.475608082201551,-0.415188442033223,-0.341860341777257,-0.324679996196445,-0.303654415835791,-0.299556410838043,0.417104897328578,0.313916655262519,-0.287196410184429,0.456719231083927,-0.301734568257508,-0.280498448839271,-0.284811969251497,0.428588459753513,0.355856100671021,0.336414034275128],[3.3790072054736,3.19228856724005,3.13383783968099,2.81858327352588,2.90703529853266,2.0048044628281,2.00645516956043,1.96833566823879,1.84904023930902,-1.87051862687113,-1.78882410513031,-1.89734684289702,1.76262376970892,-1.75494749570322,-1.71294755602572,-1.84548381088122,-1.71443610197634,-1.57435994329004,-1.55309889180495,-1.46538289201642,-1.44897032182562,1.67790560545043,1.41218843661326,-1.37255662848159,1.59710016346702,-1.42681373058886,-1.3398917157903,-1.36107937507975,1.56270634839193,1.46072633505229,1.37311272374134],[1e-50,5.27087051752225e-42,9.5658136204063e-40,5.7452377554573e-27,5.78502394502154e-18,1.9402554648469e-07,2.17424041111258e-07,5.91311787237937e-07,7.14530280407158e-07,1.44989444945863e-06,7.30462796071419e-06,1.73433620993503e-05,2.14861944062775e-05,4.14868989698182e-05,0.000292808528700162,0.000570219723948847,0.00116099628347144,0.00127643625483933,0.00195532921149267,0.00271185763444029,0.00272303438786664,0.0031753177375024,0.00957035274752194,0.0106486009221205,0.0118363379165241,0.0131170288473782,0.0135439144015828,0.0143895858713072,0.0157286555011739,0.0157545302755447,0.0224653640955143],[5e-49,1.31771762938056e-40,1.59430227006772e-38,7.18154719432163e-26,5.78502394502154e-17,1.55302886508041e-06,1.55302886508041e-06,3.69569867023711e-06,3.96961266892866e-06,7.24947224729316e-06,3.32028543668827e-05,7.22640087472927e-05,8.26392092549133e-05,0.000148167496320779,0.000976028429000539,0.00178193663734015,0.00341469495138658,0.0035456562634426,0.00514560318813861,0.0064834152092063,0.0064834152092063,0.00721663122159637,0.0208051146685259,0.0221845852544177,0.0236726758330483,0.025081322965894,0.025081322965894,0.0256956890559058,0.0262575504592412,0.0262575504592412,0.0362344582185714],[1.68421052631579e-49,4.43862780422926e-41,5.37028133075441e-39,2.41904747598202e-26,1.94863964463883e-17,5.2312551244814e-07,5.2312551244814e-07,1.24486692050092e-06,1.33713268848123e-06,2.44192749382506e-06,1.11841193656868e-05,2.43415608411933e-05,2.78363652227076e-05,4.99090513922625e-05,0.000328767470821234,0.000600231288367208,0.00115021303625653,0.00119432632031751,0.00173325581074143,0.00218388722836423,0.00218388722836423,0.00243086525359036,0.00700803862518769,0.00747270240148807,0.00797395396481627,0.00844844563061693,0.00844844563061693,0.00865538999777879,0.0088446485757444,0.0088446485757444,0.0122052911894135],[3536,3603,3775,3946,2284,3346,3742,3662,3592,4038,2361,2528,3852,4324,2666,4587,2505,3839,2233,2559,3594,4529,4500,4077,3044,4206,2027,3563,5308,5216,3303],["tags=80%, list=18%, signal=66%","tags=68%, list=19%, signal=56%","tags=72%, list=20%, signal=58%","tags=58%, list=20%, signal=47%","tags=74%, list=12%, signal=66%","tags=41%, list=17%, signal=34%","tags=39%, list=19%, signal=31%","tags=39%, list=19%, signal=32%","tags=34%, list=19%, signal=28%","tags=39%, list=21%, signal=31%","tags=27%, list=12%, signal=24%","tags=33%, list=13%, signal=29%","tags=37%, list=20%, signal=30%","tags=40%, list=22%, signal=31%","tags=33%, list=14%, signal=29%","tags=48%, list=24%, signal=37%","tags=27%, list=13%, signal=23%","tags=34%, list=20%, signal=28%","tags=25%, list=12%, signal=22%","tags=27%, list=13%, signal=24%","tags=28%, list=19%, signal=23%","tags=41%, list=23%, signal=31%","tags=35%, list=23%, signal=27%","tags=29%, list=21%, signal=23%","tags=31%, list=16%, signal=26%","tags=35%, list=22%, signal=27%","tags=40%, list=11%, signal=36%","tags=28%, list=18%, signal=23%","tags=62%, list=28%, signal=45%","tags=42%, list=27%, signal=31%","tags=26%, list=17%, signal=22%"],["MCM3/MMS22L/NOP56/UBR7/CNOT9/MCM6/UNG/PRIM2/SNRPB/RFC3/TIMELESS/DDX39A/BARD1/CKS2/RFC2/CSE1L/GINS1/SUV39H1/ATAD2/DCLRE1B/MCM7/TUBB/ORC6/TUBG1/POLD3/PSMC3IP/CTPS1/NUP205/PRKDC/EZH2/CDC25A/HELLS/NME1/BRCA2/SYNCRIP/NASP/HMGB3/LYAR/POLD1/RNASEH2A/DNMT1/MCM5/RAD51C/SMC4/KPNA2/MCM4/RAN/BRMS1L/USP1/PCNA/EIF2S1/PA2G4/TBRG4/PRPS1/CHEK2/LMNB1/TP53/DSCC1/MAD2L1/SMC1A/POLA2/GINS4/RACGAP1/CHEK1/RANBP1/TIPIN/MSH2/MYC/CDCA8/CCNE1/NUP153/DEK/CDK4/MCM2/ORC2/SLBP/BRCA1/TRIP13/POLE/STMN1/RBBP7/TCF19/GINS3/ASF1A/ASF1B/CKS1B/EED/UBE2T/AURKA/POLD2/ANP32E/XRCC6/HMGB2/POP7/TMPO/PLK4/MLH1/PPP1R8/UBE2S/H2AZ1/TACC3/CENPM/TRA2B/KIF22/XPO1/MELK/LBR/PAICS/NAA38/SRSF2/ZW10/RAD51AP1/DCK/RPA1/HNRNPD/HUS1/PLK1/PTTG1/DIAPH3/DCTPP1/EXOSC8/PDS5B/CDK1/CDKN3/NCAPD2/POLE4/SPAG5/CIT/PPM1D/ILF3/JPT1/RAD1/CBX5/SPC24/H2AX/RPA2/NUP107/E2F8/ING3/SSRP1/TK1/LIG1/SMC3/CDC20/BUB1B/WEE1/RPA3/BIRC5/AURKB/PNN/ESPL1/NOLC1/MYBL2/KIF18B/KIF2C/CDCA3/TOP2A/PRDX4/CDC25B","HDAC2/NOP56/CCT5/MCM6/C1QBP/SET/SNRPD1/DDX18/AIMP2/POLE3/CYC1/ILF2/MCM7/HSPE1/RRP9/CTPS1/SSB/NME1/SNRPA/SYNCRIP/PHB/NOP16/MCM5/PSMC4/KPNB1/KPNA2/MCM4/HNRNPR/NCBP1/RAN/USP1/SRM/PCNA/PSMD14/EIF2S1/PA2G4/EIF1AX/ETF1/MAD2L1/CCT2/HNRNPA2B1/TARDBP/RANBP1/MYC/GNL3/TFDP1/DEK/PPIA/CDK4/MCM2/RFC4/SNRPG/ORC2/RRM1/SSBP1/HNRNPU/CDK2/HNRNPA3/SERBP1/EIF3J/PSMD1/PSMB3/GLO1/ERH/EIF3B/VBP1/STARD7/EIF4E/SRPK1/TYMS/EIF4A1/U2AF1/POLD2/FAM120A/PRPS2/XRCC6/TCP1/CCT3/PSMA4/LSM2/RAD23B/H2AZ1/EXOSC7/SMARCC1/TRA2B/XPO1/YWHAE/PSMA2/CCT7/SRSF2/CSTF2/PSMA6/PSMB2/TRIM28/RUVBL2/HDDC2/ABCE1/SRSF3/PRPF31/GOT2/HNRNPD/HSPD1/CBX3/HNRNPC/CUL1/DHX15/PSMA7/VDAC1/MRPL9/G3BP1/ODC1/CCNA2/UBE2L3/HDGF/NHP2/COPS5/VDAC3/PSMD3/UBE2E1/PWP1/SF3B3/SNRPD2/CDC20/LSM7/PSMA1/SNRPD3/PSMD7/CDC45/EIF4H/NOLC1/HPRT1/EEF1B2/SNRPB2/PRDX4/CCT4/SRSF7/MRPL23","SMC2/RBL1/NCL/TRAIP/CHAF1A/MCM3/MCM6/NSD2/PRIM2/E2F3/DTYMK/DDX39A/BARD1/CASP8AP2/CKS2/SNRPD1/SUV39H1/ORC6/EZH2/CDC25A/BRCA2/SYNCRIP/NASP/HMGB3/AMD1/MCM5/SMC4/KPNB1/KPNA2/CDC6/INCENP/LMNB1/MAD2L1/SMC1A/POLA2/H2AZ2/RACGAP1/CHEK1/FBXO5/MYC/DKC1/MAP3K20/ORC5/TFDP1/DBF4/E2F1/CDK4/MCM2/MT2A/POLE/STMN1/HNRNPU/ODF2/RAD54L/KIF20B/CKS1B/AURKA/CDC7/DMD/PRMT5/CBX1/TMPO/PLK4/UBE2S/RAD23B/KATNA1/H2AZ1/TACC3/ABL1/SMARCC1/TRA2B/NUP50/KIF22/CUL3/XPO1/LBR/NUP98/SRSF2/CCNF/NUSAP1/E2F2/FANCC/HMGN2/GINS2/CDC27/SFPQ/CUL4A/HNRNPD/HUS1/TTK/PLK1/PTTG1/SAP30/KNL1/CUL1/TNPO2/PDS5B/CDK1/TENT4A/CDKN3/PRC1/HIRA/G3BP1/ODC1/CCNA2/HSPA8/UCK2/ILF3/JPT1/H2AX/KIF11/RPA2/POLQ/RPS6KA5/EXO1/WRN/TPX2/UBE2C/BUB1/CDC20/EWSR1/PBK/CDC45/BIRC5/UPF1/STIL/AURKB/NEK2/MARCKS/ESPL1/NOLC1/MYBL2/MTF2/KIF2C/TOP2A/KMT5A/CDC25B/ATF5/NDC80/KIF15/CENPE/KIF4A","NDUFA8/PMPCA/FXN/NDUFA4/NDUFS6/ATP5MC3/CYC1/PDHB/CYCS/SUCLA2/NDUFB6/UQCRC1/FH/MPC1/COX7A2/TIMM50/TIMM17A/ATP5MC1/MGST3/MRPS22/UQCRQ/SDHA/MRPS30/NDUFB3/ATP5MF/DLD/MRPS12/TIMM10/HCCS/MTX2/ATP5PF/AIFM1/MRPS15/NDUFA9/HSD17B10/ATP5F1B/NDUFA6/COX11/SLC25A11/NDUFS1/ATP5PD/MRPL15/NDUFB2/ACAT1/COX6C/MFN2/ISCA1/SDHB/NDUFC2/GRPEL1/IDH3A/COX7B/TIMM8B/AFG3L2/GOT2/COX6A1/NQO2/MTRR/COX10/PDHA1/COX5B/SLC25A4/UQCR10/ATP5PO/UQCRFS1/OPA1/SURF1/DLAT/ATP5PB/NDUFS4/UQCR11/IDH3B/TIMM13/VDAC1/OGDH/TOMM22/SDHC/ATP5ME/NDUFS3/NDUFS8/SUCLG1/ACAA1/IDH1/CASP7/VDAC3/MRPL34/MRPS11/NDUFB7/BAX/MRPL11/DLST/UQCRH/POLR2F/HSPA9/COX8A/NNT/MDH2/MDH1/NDUFB4/ATP6V1G1/IMMT/ATP5F1A/ATP5F1E/ATP6V0B/ACADM/PDHX/NDUFA5/NDUFA2/FDX1/NDUFV1/NDUFS7/MRPL35/PRDX3/COX6B1/ATP5F1D/NDUFB1","RABEPK/NOP56/UNG/NDUFAF4/BYSL/DDX18/AIMP2/PUS1/HSPE1/RRP9/TCOF1/PHB/NOP16/MRTO4/MCM5/MCM4/SRM/IMP4/PA2G4/SORD/TBRG4/UTP20/MYC/PRMT3/GNL3/CDK4/PPAN/NOP2/FARSA/PLK4/SLC19A1/MYBBP1A/IPO4/GRWD1/NOC4L/TFB2M/PES1/HSPD1/CBX3/PLK1/DCTPP1/NIP7/DUSP2","POLR2D/ALYREF/RFC3/RFC5/RFC2/POLD3/SNAPC4/FEN1/NME1/POLD1/PCNA/UMPS/TP53/NFX1/POLA1/POLA2/GTF3C5/POLR1C/RFC4/UPF3B/NELFB/REV3L/TYMS/RAD51/SF3A3/PRIM1/MPC2/TAF9/POLR3C/ADRM1/AK1/NUDT9/GTF2H1/POLR2E/SURF1/PDE4B/POLE4/EDF1/ZWINT/POLH/POLR2A/RPA2/RAE1/SAC3D1/ERCC8/DGCR8/SSRP1/GTF2A2/LIG1/SDCBP/GTF2H3/GTF2H5/POLR2F/POLR2I/ITPA/RPA3/PNP/RALA/ERCC3/HPRT1/TAF6","HMGCS2/HSP90AA1/BCKDHB/GRHPR/PDHB/SUCLA2/HSDL2/FH/NCAPH2/HADH/SDHA/DLD/CEL/HCCS/MIX23/YWHAH/FABP1/ALDH1A1/ME1/REEP6/HSD17B10/H2AZ1/UGDH/EHHADH/GCDH/ACAT2/PDHA1/ECI2/ALDH9A1/ADSL/ALDH3A2/CYP1A1/IDH3B/SMS/HSD17B4/METAP1/ODC1/SDHC/SUCLG1/HIBCH/ACAA1/S100A10/VNN1/IDH1/HMGCS1/CRAT/ADH1C/DLST/PTS/CBR3/MDH2/MDH1/ACSS1/KMT5A/GPD2/ACADM","CASP3/IGFBP2/POLE3/CTSV/SIGMAR1/FEN1/DNAJA1/AMD1/GAL/CDC5L/CCNE1/RFC4/TARS1/HNRNPU/CDK2/AP2S1/PPIF/SQSTM1/PRPF3/GGH/PPAT/STK25/GRPEL1/CCND3/CHRNA5/PDAP1/ARRB2/SLC25A4/GCH1/OLFM1/CYP1A1/PSMC3/CNP/NR4A1/FKBP4/HMOX1/YKT6/BID/NAT1/TGFBRAP1/H2AX/ACAA1/HYAL2/STIP1/CXCL2/LYN/TUBA4A/RASGRP1/BSG/CDC34/NUP58/TCHH/TYRO3/CYB5B/MRPL23/ATF3","TOMM40/UNG/CACYBP/DDX39A/EEF1E1/PSMG1/HSPE1/TUBG1/RRP9/NUP205/PRDX1/CDC25A/ABCF2/PSMC4/MCM4/PSMD14/GLA/SORD/IFI30/ETF1/ATP5MC1/PNO1/ARPC5L/PPIA/MCM2/HSPA4/SQSTM1/AURKA/UCHL5/PSMD12/ME1/PSMA4/DHFR/PSMA3/PITPNB/PSME3/CCNF/CCT6A/RPA1/NIBAN1/HSPD1/PLK1/PDAP1/HMBS/GTF2H1/PSMC2/GSR/POLR3G/YKT6/PSAT1/GCLC/COPS5/IDH1/HMGCS1/STIP1/BUB1/ACTR3/TUBA4A/NUFIP1/HSPA9/PNP/QDPR/HPRT1/PSMB5/FDXR/GOT1/CYB5B/IMMT","PPARGC1A/XPNPEP1/PPP1R15A/HOXB9/ANXA2/HS3ST1/SULT2B1/RORA/TPBG/KIF5A/SDC4/WSB1/VEGFA/EDN2/STC1/NR3C1/TES/CCN2/HEXA/BGN/CDKN1C/SLC37A4/PGK1/LDHA/CHST3/CAV1/GCNT2/KDM3A/ANGPTL4/HK1/SLC2A1/NDST2/MAFF/SDC3/HK2/PLAUR/BHLHE40/ANKZF1/EFNA3/CAVIN3/PGAM2/PDGFB/ERO1A/PNRC1/CA12/ERRFI1/INHA/IGFBP1/P4HA2/ADM/PPFIA4/PFKFB3/CDKN1B/IDS/KLHL24/CP/KDELR3/ALDOA/CCNG2/GAA/VLDLR/BNIP3L/AMPD3/MXI1/EFNA1/ALDOC/P4HA1/PRKCA/ENO2/TGM2/DDIT4/PFKP","IFITM3/XAF1/IL15/PSMB9/APOL6/IFITM2/IFIT3/TRAFD1/IFI27/PSMB8/TAPBP/B2M/UBE2L6/PSMB10/C1S/NCOA3/RNF213/IFI44/IFI44L/RBCK1/VAMP8/IRF5/CIITA/TXNIP/IFIT2/PARP14/SOCS3/CASP4/IL4R/MVP/IFNAR2/C1R/DHX58/IL15RA/ST3GAL5/VAMP5/CD40/TNFSF10/IRF9/BATF2/STAT2/PML/CFB/GBP4/IFIT1/PFKP/IRF7/BST2","EIF2AK2/IFITM3/IL15/PSMB9/CD47/IFITM2/IFIT3/TRAFD1/IFI27/PSMB8/TMEM140/B2M/UBE2L6/C1S/IFI44/IFI44L/IFITM1/TXNIP/IFIT2/TRIM5/PARP14/IL4R/DHX58/UBA7/IRF9/BATF2/STAT2/GBP4/IRF7/GBP2/BST2","CDK5RAP2/CENPJ/BRCA2/MAPRE1/SMC4/INCENP/LMNB1/SMC1A/ECT2/CEP72/RACGAP1/FBXO5/KNTC1/RABGAP1/GEMIN4/PPP4R2/TBCD/TRIO/KIF20B/AURKA/DOCK4/CKAP5/KATNA1/ABL1/PALLD/KIF22/YWHAE/CSNK1D/NUSAP1/SASS6/ALS2/CDC27/EZR/EPB41L2/TTK/PLK1/TSC1/CNTROB/CDK1/PRC1/TUBGCP3/CEP131/MYH10/KIF11/PCNT/LATS1/SAC3D1/CEP192/TPX2/BUB1/NEDD9/SMC3/TUBA4A/WASF1/ANLN/DYNLL2/ITSN1/BIRC5/NEK2/MARCKS/ESPL1/TIAM1/MID1/KIF2C/ARHGEF11/ARHGDIA/TOP2A/NDC80/KIF15/CENPE/CAPZB/KIF4A/KIF23","HBEGF/KCNH2/PYGM/KLF5/FGF2/SLC6A8/TNNT2/SSPN/TNNI2/LSP1/CKB/COL6A2/BDKRB2/MEF2C/MYO1C/MEF2D/CACNG1/SOD3/DENND2B/WWTR1/SIRT2/PRNP/TSC2/GABARAPL2/ITGA7/ADCY9/DAPK2/PTP4A3/CHRNB1/LDB3/BHLHE40/PLXNB2/MYOM1/AGRN/PGAM2/ITGB5/EIF4A2/MEF2A/SPEG/SCD/ITGB1/PPFIA4/GADD45B/ITGB4/TGFB1/GAA/FKBP1B/TPM2/LPIN1/SH2B1/CTF1/OCEL1/TNNT1/FOXO4/MAPRE3/SORBS3/HDAC5/ERBB3","PROC/RGN/HPN/MMP10/S100A13/SIRT2/CAPN2/CTSO/MAFF/MMP11/CAPN5/HTRA1/ITGA2/MMP15/CTSB/MMP7/PROS1/PRSS23/C1S/PDGFB/F10/MMP14/MST1/LGMN/MMP1/LRP1/TMPRSS6/ANG/THBD/DUSP6/C1R/TIMP1/TF/RAPGEF3/CFB/CRIP2","LEFTY2/BMP2/RAB31/PPP1R15A/LTBP2/NOG/CDKN1C/SMURF1/XIAP/WWTR1/SMAD6/SKIL/BCAR3/CDH1/ACVR1/ID3/BMPR2/KLF10/SMAD7/JUNB/PMEPA1/IFNGR2/SMAD3/TGFB1/ID2/ID1","CHKA/STARD4/TP53INP1/LGALS3/PLAUR/HSD17B7/SREBF2/FADS2/PNRC1/GSTM2/ERRFI1/TRIB3/SCD/LGMN/STX5/ALDOC/ANTXR2/SEMA3B/LDLR","ANXA2/CELF2/ATP2B4/RASA2/INSIG1/HAS2/ATRN/MET/NR3C1/NOTCH2/PTPN21/MAGI2/MAP2K5/PIK3CD/IRS1/INPP4B/DBP/CAV1/RUNX1/PTEN/TFPI/RBPMS/WDR37/SLC22A18/PTPRM/BHLHE40/IGF1R/CDON/MAP1B/NFIB/SMAD7/FHL2/ATXN1/SMAD3/MGLL/CDKN1B/NRP1/MGMT/VLDLR/ERBB2/LTBP1/TGFBR3/ID1/ZMIZ1/PRKCA/LDLR","CLDN9/NF1/INPPL1/PTEN/SDC3/CLDN4/TSPAN4/ITGA2/COL17A1/LAMB3/PIK3CB/CDH1/LAMC2/MSN/TIAL1/SGCE/TRAF1/CDH15/CDSN/ITGA10/SLIT2/CDH3/FLNC/RAC2/SPEG/NLGN2/GRB7/RHOF/ITGB1/THBS3/SRC/ITGB4/BAIAP2/TAOK2/CALB2/ITGA3/RRAS/PCDH1/MPZL2/SORBS3/FSCN1/CX3CL1","CHPF/GALE/TPST1/MIOX/SLC37A4/PGK1/LDHA/PKM/CLDN9/ANGPTL4/SDC3/CAPN5/B3GAT3/MPI/HK2/TFF3/ANKZF1/EFNA3/AGRN/PGAM2/GAL3ST1/ERO1A/B4GALT4/RBCK1/CHPF2/CHST6/P4HA2/PLOD2/IL13RA1/PGLS/PPFIA4/NT5E/SLC16A3/KDELR3/ANG/B3GNT3/ALDOA/VLDLR/B3GAT1/MXI1/TGFA/NOL3/P4HA1/EGLN3/SPAG4/ENO2/SDC1/DDIT4/LHPP/IDUA/PFKP","SULT2B1/SYNGR1/TPBG/FARP1/TGIF2/WWC1/MUC1/ESRP2/ANXA9/TJP3/KDM4B/MAST4/NADSYN1/AFF1/MINDY1/ADCY9/SLC2A1/RAPGEFL1/MAPT/ABAT/THSD4/CELSR1/CLIC3/BHLHE40/TTC39A/KAZN/TFF3/IGF1R/PRSS23/DHRS3/ENDOD1/KLF10/KRT19/CA12/ALDH3B1/FHL2/MYOF/RARA/PEX11A/KRT15/TMPRSS3/PLAAT3/KCNK15/SYT12/INPP5F/ASB13/CALB2/AMFR/TFF1/SCNN1A/TGM2/KLK10/SEMA3B/ABCA3","TSN/STRBP/PSMG1/EZH2/DBF4/RFC4/AURKA/TOPBP1/MTOR/YBX2/HSPA4L/TTK/IP6K1/CDK1/CDKN3/PRKAR2A/PEBP1/MEP1B/NCAPH/DMC1/VDAC3/BUB1/RPL39L/NEK2/NF2/KIF2C/AGFG1/SLC12A2/MAP7/CCNB2/GMCL1/CLVS1/STAM2/COIL","CYC1/MIGA2/UQCRC1/HADH/MGST3/UQCRQ/COQ3/DLD/AIFM1/UCK1/MTCH2/ME1/ABCB8/REEP6/SLC19A1/MRPL15/SDHB/GRPEL1/IDH3A/ABCA1/COX7B/COX6A1/UQCR10/FAH/ATP5PO/DLAT/SORBS1/UQCR11/SLC25A10/SOD1/SDHC/NDUFS3/PPM1B/SUCLG1/HIBCH/UBQLN1/IDH1/NDUFB7/CRAT/DBT/COX8A/ITSN1/MDH2/QDPR/ENPP2/IFNGR1/GPAM/IMMT/GPD2/DHRS7B/ACADM/NDUFA5/DNAJC15/ESYT1/PRDX3/SAMM50/NKIRAS1/PPP1R15B/GPX3/PTCD3/MCCC1/NDUFAB1/CHUK/ECHS1","RELA/LCK/AHR/PIK3R5/LIF/CD48/PTAFR/STAB1/HAS2/C5AR1/TPBG/IL10RA/SCARF1/MET/TNFSF9/SCN1B/HPN/RTP4/MXD1/SELENOS/EREG/EIF2AK2/CD14/IL15/EMP3/GABBR1/IL18/AXL/BDKRB1/TAPBP/PLAUR/IFITM1/MMP14/OSMR/P2RX4/ADM/IFNGR2/IL4R/IL15RA/CD82/TIMP1/CD40/TNFSF10/CX3CL1/PTPRE/IRF7/BST2/LDLR","HDAC2/CCND2/DLL1/SKP2/TP53/MYC/NOTCH1/LEF1/CUL1/FZD1/NKD1/TCF7","QSOX1/DKK1/CRLF1/PPIB/CD59/FGF2/SLC6A8/CD44/ACTA2/FUCA1/VCAN/SDC4/COL6A2/VEGFA/NOTCH2/FBLN2/PDLIM4/CCN2/PCOLCE2/BGN/APLP1/NNMT/IL15/EMP3/HTRA1/ITGA2/ABI3BP/LAMC2/PLAUR/ITGB5/CADM1/SLIT2/FERMT2/MMP14/PLOD2/VIM/ITGB1/PMEPA1/MMP1/GADD45B/LRP1/LGALS1/CAPG/NT5E/TGFB1/TPM2/EFEMP2/ID2/PCOLCE/TIMP1/TGFBR3/ENO2/SDC1/TGM2","CPM/GZMA/ADRA2B/PLEK/APOBEC3F/PDP1/ZEB1/FCN1/CD46/CD40LG/CTSH/SERPING1/CCL5/NOTCH4/DOCK10/WAS/LTA4H/PRCP/JAK2/CD59/PRKCD/XPNPEP1/PRSS36/LCK/SPOCK2/PIK3CA/PIK3R5/S100A9/EHD1/DOCK9/S100A13/STX4/PSMB9/CTSO/MAFF/LGALS3/MMP13/DGKH/MMP15/CTSB/PPP4C/PLAUR/DUSP5/C1S/PDGFB/F10/CDA/KCNIP2/HSPA1A/PCSK9/MMP14/APOBEC3G/F5/LGMN/LIPA/SRC/CASP4/DGKG/LRP1/CP/TMPRSS6/TIMP2/ANG/DUSP6/KLK1/C1R/TIMP1/CFB/CBLB/IRF7","CLCN3/TSPAN5/NR3C1/ADD2/BLVRA/OSBP2/TMCC2/BACH1/SLC30A1/SLC10A3/DMTN/PPOX/RANBP10/SLC25A37/EZH1/SMOX/NARF/RIOK3/TNRC6B/SEC14L1/ALDH1L1/SLC2A1/PDZK1IP1/BTRC/CTSB/HAGH/NFE2/CDR2/ENDOD1/MARCHF2/MINPP1/EPOR/NUDT4/OPTN/P4HA2/PGLS/RAP1GAP/BCAM/ARHGEF12/LRP10/DCAF11/BNIP3L/MXI1/PPP2R5B/CAST/TNS1/MARCHF8/YPEL5/KHNYN","PRDX1/TXN/PTPA/NDUFA6/GCLM/GLRX2/STK25/SRXN1/MGST1/GSR/ATOX1/SOD1/GCLC/SCAF4/ABCC1/PRDX4/NDUFB4/EGLN2/PRDX2/FTL/NQO1/GPX3/SOD2/OXSR1/NDUFS2/ERCC2/TXNRD1/MSRA/HHEX/TXNRD2","VAV3/MAPKAP1/E2F1/CDK4/CDK2/PFN1/EIF4E/SQSTM1/TRAF2/CAB39/UBE2N/RAF1/AKT1S1/PPP2R1B/PLCG1/HRAS/CDK1/PRKAR2A/MAP3K7/ACTR3/CSNK2B/CLTC/RPTOR/AKT1/TIAM1/ARHGDIA/PTPN11/PIN1/CFL1/DUSP3/THEM4/NFKBIB/CAB39L/IL2RG/YWHAB/ITPR2/MAPK1/MKNK1/TBK1/PRKAA2/PLA2G12A","SGK1/XRCC5/PA2G4/SORD/FKBP5/SRP19/CENPN/XRCC6/ABCC4/PLPP1/NGLY1/B4GALT1/ADRM1/CCND3/GSR/GNAI3/SMS/MAF/DBI/TPD52/GUCY1A1/HMGCS1/TMPRSS2/RRP12/AKT1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ID<\/th>\n      <th>Description<\/th>\n      <th>setSize<\/th>\n      <th>enrichmentScore<\/th>\n      <th>NES<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>qvalues<\/th>\n      <th>rank<\/th>\n      <th>leading_edge<\/th>\n      <th>core_enrichment<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><img src="gsea_files/figure-html/unnamed-chunk-31-1.png" width="864" style="display: block; margin: auto;" /><img src="gsea_files/figure-html/unnamed-chunk-31-2.png" width="864" style="display: block; margin: auto;" /> <img src="gsea_files/figure-html/unnamed-chunk-32-1.png" width="576" style="display: block; margin: auto;" /></p>
<p><img src="gsea_files/figure-html/unnamed-chunk-33-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>TO- check :</p>
<p>Q: In scRNA-seq sometimes you encounter a lot of Mitochondrial and Ribosomal RNA genes and usually you set filters to filter them out since they might indicate degradation. When you do an enrichment in bulk and there is a huge enrichment in GO terms for rRNA genes, could it mask your other highly differentiated genes in your analysis? Does it mean the sample has degradation and should be excluded? Or how could you handle this? (Alicia) A: If mitochondrial or ribosomal RNA GO terms are enriched in bulk RNA-seq datasets, this would not “mask” other gene sets from also being found as significantly enriched. What could happen is that if you plot only the top “n” (e.g. 10) enriched terms and mitochondrial/rRNA terms are the most significant terms, then you would perhaps miss out on plotting other interesting gene sets. You could increase the number of gene sets plotted by changing the parameter “showCategory” to a higher value (e.g. showCategory = 20). Otherwise, you could also obtain a results table for all gene sets, regardless of them being significant or not and check for particular gene sets of interest there. Yes, I agree here. In addition, I think that when genes linked to ribosomal proteins or mitochondrial proteins are enriched in a bulk RNA seq, this might have biological relevance. Eg a sample might be more translationally active than another, so gene sets related to ribosomal proteins will come up. The same for the mitochondrial proteins which can involve impacts on mitochondrial dysfunctionality or oxidative phosphorylation. So it might be very relevant in certain biological systems (eg in cardiac cells, I think that mitochondria are very active <a href="https://elifesciences.org/articles/30952" class="uri">https://elifesciences.org/articles/30952</a>) That said, I once analyzed an RNA seq data set where most of the DE genes were coding for ribosomal proteins, and we suspected a problem with gene capture but we were never sure about it. When we plotted the fold change of the genes against gene length, we saw that the very short genes were highly up-regulated, while the very long genes were strongly down-regulated. The fold change was correlated with the gene length. So in the end we only had ribosomal proteins that came out as significant, and very few other genes. We therefore suspected a problem with the RNA seq capture or extraction.</p>
<p>– &gt; Plot gene length against logFC</p>
</div>
</div>


<p>Copyright &copy; 2022 Alicia Pliego. FTP lab. All rights reserved.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
