---
title: "Enrichment analysis"
author: "Alicia Pliego"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_collapsed: true
    toc_depth: 3
    number_sections: true
    theme: sandstone

---

```{r setup, include=FALSE, echo=FALSE, message=FALSE}
library("DESeq2")
library(tximport)
library(plotly)
library(pheatmap)
library(org.Hs.eg.db)
library(DOSE)
library(pathview)
library(clusterProfiler)
library(AnnotationHub)
library(ensembldb)
library(tidyverse)
  library(EnhancedVolcano)
library(ComplexHeatmap)
library(DT)
library(enrichplot)
library(org.Hs.eg.db)

# set seed
set.seed(1234)
```


# Gene Enrichment analysis

```{r,  echo=FALSE, warning=FALSE, message=FALSE}
#Import quantification files generated by DRAGEN from SALMON.
dirs <- list.files("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/data_v2/")
```

```{r,echo=FALSE, warning=FALSE, message=FALSE}
quant_files <- list.files("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/data_v4/",pattern="quant.genes.sf",recursive = TRUE,full.names = TRUE)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# txi import quant files by genes from salmon into DESEQ2
txi <- tximport(quant_files,type="salmon", txOut=TRUE)
```

Samples to compare in the Differential Analysis:
Excluding sample 31, 35 and 37 from the analysis, a Differential expression analysis was performed:

```{r,  echo=FALSE, warning=FALSE, message=FALSE}
metadata = read.table('/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/BRAF_samples_v4.txt', header = TRUE)

```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeqDataSetFromTximport(txi, 
                                colData = metadata,
                                design <- ~Treatment)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds) >= 10
keep <- rowSums(assay(dds) >= 10) >= 2
dds <- dds[keep,]
```

```{r results , echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeq(dds)
res <- results(dds, contrast = c("Treatment", "Treated", "Untreated"))
```

```{r,  echo=FALSE, warning=FALSE, message=FALSE}
# Help: hypergeometric test (like one-sided Fisher test = greater)

# Test 3 gene sets among the genes up-regulated in NK cells,
# with enricher()
# First, obtain the genes up-regulated in NK:
res<- as.data.frame(res)
res$symbol <- rownames(res)
resistant_up_genes<-subset(res, res$log2FoldChange>0&res$padj<=0.05)$symbol

# Import wnt gene set:
wnt<-read.gmt("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/wnt_signature_own.gmt")

hyper_3genesets<-enricher(gene=resistant_up_genes,
                      universe = res$symbol,
                      TERM2GENE = wnt,
                      maxGSSize = 1000)
```



## WNT Signaling pathway enrichment

### Volcano plot of the DE genes belonging to the WNT pathway
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 14, fig.align='center'}

library(ggplot2)
library(ggrepel)

sig_genes<-subset(res, res$symbol %in% wnt$gene)
sig_genes_label<-subset(sig_genes, sig_genes$padj<=0.05)

ggplot(res, aes(x = log2FoldChange,  # 
                     y = -log10(padj))) +
        geom_point(color="grey87")  +
        ggtitle("Genes belonging to the Basal gene signature") +
        theme_bw() +
        geom_text_repel(data = sig_genes_label, 
                        aes(x=log2FoldChange,
                            y=-log10(padj),label=symbol),
                        max.overlaps = 20) +
        geom_point(data=sig_genes, col="dodgerblue2") +
        theme(legend.position = "none") +
        scale_x_continuous(name = "log2(fold change) Treated vs non-Treated") +
        scale_y_continuous(name = "-log10 p-value") +
        geom_hline(yintercept = -log10(0.05), linetype="dashed") +
        geom_vline(xintercept = 0, linetype="dashed")
```

We can see an enrichment in the treated BRAF cases (right) for positive regulators of the WNT signaling cascade, whilst on the left (downregulated for the treated samples) we can see an enrichment for the suppressor genes. 
From these results we can say it seems there is an upregulation of activator genes in the treated samples and downregulation of suppressor genes of the WNT signaling.


```{r, echo=FALSE, warning=FALSE, message=FALSE}
vst <- vst(dds, blind=FALSE)
vst <- assay(vst)
vst <- as.data.frame(vst)
vst_sig <- vst[rownames(vst) %in% rownames(sig_genes_label),]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

names(vst_sig)[1] <- "LR2022_26"
names(vst_sig)[2] <- "LR2022_27"
names(vst_sig)[3] <- "LR2022_28"
names(vst_sig)[4] <- "LR2022_29"
names(vst_sig)[5] <- "LR2022_30"
names(vst_sig)[6] <- "LR2022_32"
names(vst_sig)[7] <- "LR2022_33"
names(vst_sig)[8] <- "LR2022_34"
names(vst_sig)[9] <- "LR2022_35"
names(vst_sig)[10] <- "LR2022_36"

```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- t(scale(t(vst_sig)))
```

### Heatmap of the genes belonging to the WNT pathway
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 6, fig.height = 6, fig.align='center'}
Heatmap(heat, 
      row_names_gp = gpar(fontsize = 8),
    top_annotation = HeatmapAnnotation(foo = anno_block(gp = gpar(fill = 2:4),
        labels = c("Untreated", "Treated"), 
        labels_gp = gpar(col = "white", fontsize = 10))),
    column_km = 2)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Import wnt gene set:
mofit<-read.gmt("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/mofit_signature.gmt")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#idType(OrgDb = "org.Hs.eg.db") # to determine allowed gene id type
sig_genes<-subset(res, res$symbol %in% mofit$gene)
sig_genes_label<-subset(sig_genes, sig_genes$padj<=0.05)
vst_sig <- vst[rownames(vst) %in% rownames(sig_genes_label),]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

names(vst_sig)[1] <- "LR2022_26"
names(vst_sig)[2] <- "LR2022_27"
names(vst_sig)[3] <- "LR2022_28"
names(vst_sig)[4] <- "LR2022_29"
names(vst_sig)[5] <- "LR2022_30"
names(vst_sig)[6] <- "LR2022_32"
names(vst_sig)[7] <- "LR2022_33"
names(vst_sig)[8] <- "LR2022_34"
names(vst_sig)[9] <- "LR2022_35"
names(vst_sig)[10] <- "LR2022_36"

```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- t(scale(t(vst_sig)))
```

### Heatmap of the genes belonging to the Mofit pathway
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 6, fig.height = 6, fig.align='center'}
Heatmap(heat, 
      row_names_gp = gpar(fontsize = 8))
```

## GO enrichment
```{r, echo=FALSE, warning=FALSE, message=FALSE}
gl<-res$stat
names(gl)<-make.names(res$symbol, unique = T)
gl<-gl[order(gl, decreasing = T)]

GO_T_U<-gseGO(gl, ont="BP",
                OrgDb = org.Hs.eg.db,
                keyType = "SYMBOL",
                minGSSize=30,
                eps=1e-50,  # or try using eps=0
                seed=T)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
GO_T_U_simplify <- clusterProfiler::simplify(GO_T_U)
#View(GO_T_U_simplify@result)
#GO_T_U_simplify@result[GO_T_U_simplify@result,]
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
#write.csv(GO_NK_Th@result, "GO_GSEA_NK_vs_Th.csv",
#          row.names = F, quote = F)

# How to obtain the list of leading edge genes for a gene set:
#unlist(strsplit(GO_T_U@result[GO_T_U@result$Description=="",11],
#                "\\/"))

```

List of GO terms differentially upregulated:

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 8, fig.align='center'}

par(mar=c(5,20,3,3))
barplot(rev(-log10(GO_T_U@result$p.adjust[1:10])),
        horiz = T, names=rev(GO_T_U@result$Description[1:10]),
        las=2, xlab="-log10(adj.p-value)",
        cex.names = 0.7,
        col="lightgreen") 
abline(v=-log10(0.05))
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 10, fig.align='center'}

    # Barplot of NES of the top GO gene sets, with different color for up- or down-reg gene sets:
    sorted_GO_T_U<- GO_T_U@result[order(GO_T_U@result$NES, decreasing = F),]
    sorted_GO_T_U$color<-ifelse(sorted_GO_T_U$NES<0, "red", "darkblue")

    par(mar=c(5,20,3,3))
    barplot(sorted_GO_T_U$NES[c(1:10, (nrow(sorted_GO_T_U)-9):nrow(sorted_GO_T_U))],
        horiz = T, names=sorted_GO_T_U$Description[c(1:10, (nrow(sorted_GO_T_U)-9):nrow(sorted_GO_T_U))],
        las=2, xlab="NES",
        cex.names = 0.7,
        col=sorted_GO_T_U$color[c(1:10, (nrow(sorted_GO_T_U)-9):nrow(sorted_GO_T_U))]) 
    abline(v=0)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
T_U_up_genes<-subset(res, res$log2FoldChange>0&res$padj<=0.05)$symbol

# Provide list of significant genes for over-representation analysis of GO gene sets 
# using enrichGO():
GO_enrich<-enrichGO(gene=T_U_up_genes,
                    OrgDb = org.Hs.eg.db,
                    keyType = "SYMBOL",
                    maxGSSize = 30)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#keytypes(org.Hs.eg.db)
# convert from= "ENSEMBL" to "SYMBOL" and "ENTREZID"
gene_convert <- bitr(as.character(res$symbol), 
                     fromType="SYMBOL", 
                     toType=c("SYMBOL", "ENTREZID"), OrgDb="org.Hs.eg.db")


```

## GSEA



```{r, echo=FALSE, warning=FALSE, message=FALSE}
library(pathview)

# pathview map with non-significant genes in grey:
# set log fold change of non-significant genes to 0:
res$logFC_0<-ifelse(res$padj>0.05, 0, res$log2FoldChange)

# create named vector of fold change values:
genePW<-res$logFC_0
names(genePW)<-res$symbol

# create pathview map of Natural killer cell mediated cytotoxicity = hsa04650
#pathview(gene.data  = genePW,
#         pathway.id = "hsa04010",
#         species    = "hsa",
#         gene.idtype = "SYMBOL")

# pathview map for Ribosome = hsa03010
#pathview(gene.data  = genePW,
#         pathway.id = "hsa04330",
#         species    = "hsa",
#         gene.idtype = "SYMBOL")

#pathview(gene.data  = genePW,
#         pathway.id = "hsa04115",
#         species    = "hsa",
#         gene.idtype = "SYMBOL")

#pathview(gene.data  = genePW,
#         pathway.id = "hsa03008",
#         species    = "hsa",
#         gene.idtype = "SYMBOL")

#pathview(gene.data  = genePW,
#         pathway.id = "hsa04110",
#         species    = "hsa",
#         gene.idtype = "SYMBOL")

#pathview(gene.data  = genePW,
#         pathway.id = "hsa04310",
#         species    = "hsa",
#         gene.idtype = "SYMBOL")
```

### Pathways
```{r , echo=FALSE, out.width = '100%'}
knitr::include_graphics("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/report/RNAseq-BRAF/images/hsa04310.pathview.png")
```
```{r , echo=FALSE, out.width = '100%'}
knitr::include_graphics("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/report/RNAseq-BRAF/images/hsa04110.pathview.png")
```
```{r , echo=FALSE, out.width = '100%'}
knitr::include_graphics("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/report/RNAseq-BRAF/images/hsa03008.pathview.png")
```
```{r , echo=FALSE, out.width = '100%'}
knitr::include_graphics("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/report/RNAseq-BRAF/images/hsa04115.pathview.png")
```
```{r , echo=FALSE, out.width = '100%'}
knitr::include_graphics("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/report/RNAseq-BRAF/images/hsa04330.pathview.png")
```
```{r , echo=FALSE, out.width = '100%'}
knitr::include_graphics("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/report/RNAseq-BRAF/images/hsa04010.pathview.png")
```

```{r , echo=FALSE, out.width = '100%'}
knitr::include_graphics("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/report/RNAseq-BRAF/images/hsa04310.pathview.png")
```


## Hallmarks MSigDB enriched pathways
```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Import hallmark, convert to term2gene and run GSEA:
term2gene_h<-read.gmt("/Users/aliciapliego/Projects/FMI_RNAseq/BRAF/data/h.all.v7.5.1.symbols.gmt")
#head(term2gene_h)
#length(unique(term2gene_h$term))

# Run GSEA with the function that allows to use custom gene sets, 
# provide the named vector of t statistics
h_T_U<-GSEA(gl, TERM2GENE = term2gene_h,
                eps=1e-50, # or try eps=0
                seed=T)

#View(h_T_U@result)

# Number of significant gene sets:
#length(which(h_T_U@result$p.adjust<=0.05)) 
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 6, fig.height = 8, fig.align='center'}

h_T_U_df <- as.data.frame(h_T_U)
DT::datatable(h_T_U_df)  
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 9, fig.height = 8, fig.align='center'}

# A dotplot with geneRatio or NES on the x-axis:
dotplot(h_T_U)
dotplot(h_T_U, x="NES", orderBy="p.adjust")
```
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 6, fig.height = 8, fig.align='center'}

# A barcode plot:
gseaplot2(h_T_U, geneSetID = "HALLMARK_WNT_BETA_CATENIN_SIGNALING",
           title="HALLMARK_WNT_BETA_CATENIN_SIGNALING")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 6, fig.height = 8, fig.align='center'}

gseaplot2(h_T_U, geneSetID = "HALLMARK_PI3K_AKT_MTOR_SIGNALING",
           title="HALLMARK_PI3K_AKT_MTOR_SIGNALING")
```


TO- check :

Q: In scRNA-seq sometimes you encounter a lot of Mitochondrial and Ribosomal RNA genes and usually you set filters to filter them out since they might indicate degradation. When you do an enrichment in bulk and there is a huge enrichment in GO terms for rRNA genes, could it mask your other highly differentiated genes in your analysis? Does it mean the sample has degradation and should be excluded? Or how could you handle this? (Alicia)
A: If mitochondrial or ribosomal RNA GO terms are enriched in bulk RNA-seq datasets, this would not “mask” other gene sets from also being found as significantly enriched. What could happen is that if you plot only the top “n” (e.g. 10) enriched terms and mitochondrial/rRNA terms are the most significant terms, then you would perhaps miss out on plotting other interesting gene sets. You could increase the number of gene sets plotted by changing the parameter “showCategory” to a higher value (e.g. showCategory = 20). Otherwise, you could also obtain a results table for all gene sets, regardless of them being significant or not and check for particular gene sets of interest there.
Yes, I agree here. In addition, I think that when genes linked to ribosomal proteins or mitochondrial proteins are enriched in a bulk RNA seq, this might have biological relevance. Eg a sample might be more translationally active than another, so gene sets related to ribosomal proteins will come up. The same for the mitochondrial proteins which can involve impacts on mitochondrial dysfunctionality or oxidative phosphorylation. So it might be very relevant in certain biological systems (eg in cardiac cells, I think that mitochondria are very active  https://elifesciences.org/articles/30952)
That said, I once analyzed an RNA seq data set where most of the DE genes were coding for ribosomal proteins, and we suspected a problem with gene capture but we were never sure about it. When we plotted the fold change of the genes against gene length, we saw that the very short genes were highly up-regulated, while the very long genes were strongly down-regulated. The fold change was correlated with the gene length. So in the end we only had ribosomal proteins that came out as significant, and very few other genes. We therefore suspected a problem with the RNA seq capture or extraction. 

-- > Plot gene length against logFC